internal/command/resourceplugin/subst.go

```diff
@@ -0,0 +1,163 @@
+package resourceplugin
+
+import (
+	"encoding/json"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/signadot/cli/internal/clio"
+	"github.com/signadot/cli/internal/config"
+	"github.com/signadot/go-sdk/models"
+)
+
+func loadResourcePlugin(file string, tplVals config.TemplateVals, forDelete bool) (*models.ResourcePlugin, error) {
+	substMap, err := substMap(tplVals)
+	if err != nil {
+		return nil, err
+	}
+	rpt, err := clio.LoadYAML[any](file)
+	if err != nil {
+		return nil, err
+	}
+	if forDelete {
+		*rpt = extractName(*rpt)
+	}
+
+	if err := substTemplate(rpt, substMap); err != nil {
+		return nil, err
+	}
+	return unstructuredToResourcePlugin(*rpt)
+}
+
+func extractName(rpt any) map[string]any {
+	topLevel, ok := rpt.(map[string]any)
+	if !ok {
+		return map[string]any{}
+	}
+	for k := range topLevel {
+		if k == "name" {
+			continue
+		}
+		delete(topLevel, k)
+	}
+	return topLevel
+}
+
```

Comment:
@jmsktm, there is a lot of code duplication between all the functions below and the ones in `internal/command/sandbox/subst.go`.

I think it would be a good idea to locate this in some common package.